docs/design/server_side_charts.md

```diff
@@ -0,0 +1,44 @@
+# Supporting charts on the server side
+As discussed in #367, the server side components manager and resourcifier have not yet been refactored, as of this writing, to work with [the new chart format](docs/design/chart_format.md). Presumably, expandybird should not be affected.
+
+## Context
+In discussions about the refactoring on the Slack channel, a number of potential requirements were discussed that could complicate matters, such as:
+* supporting multiple templating technologies within a single chart.
+* supporting multiple templating technologies within a single dependency graph.
```

Comment:
By dependency you mean "on another chart" right?  In Terraform this is called a module, and in Cloud Formation a "sub stack".

This differs from "references" in both CF and Terraform which create ordering dependencies at deployment time and allow one resource to be configured using generated attributes (e.g. addresses, uuids, etc) of other resources.  I am not certain Kubernetes (as opposed to all IAAS providers) needs the concept of "reference" as the links between resources are "soft" and resources can be deployed in any order.  That is good, as it simplifies things considerably!

It also differs from the imports that you can do in e.g. Python / Jinja / YAML which are resolved by those interpreters and Helm needs not care about them (other than ensuring that all the required files are provided).

I suppose at least for the first pass, we will not need this kind of char nesting functionality at all, although it will be surely needed for real applications of even modest complexity (e.g. guestbook).  When we add it, I suspect it will be necessary to support multiple templating technologies, as different charts may be maintained by different organizations.

So if we want to cut risk, having no dependencies at first seems reasonable.